containers/src/Kokkos_DynRankView.hpp

```diff
@@ -1656,6 +1666,24 @@ struct DynRankViewRemap {
     Kokkos::parallel_for("Kokkos::DynRankViewRemap", Policy(0, n0), *this);
   }
 
+  DynRankViewRemap(ExecSpace const& exec_space, const OutputView& arg_out,
+                   const InputView& arg_in)
+      : output(arg_out),
+        input(arg_in),
+        n0(std::min((size_t)arg_out.extent(0), (size_t)arg_in.extent(0))),
+        n1(std::min((size_t)arg_out.extent(1), (size_t)arg_in.extent(1))),
+        n2(std::min((size_t)arg_out.extent(2), (size_t)arg_in.extent(2))),
+        n3(std::min((size_t)arg_out.extent(3), (size_t)arg_in.extent(3))),
+        n4(std::min((size_t)arg_out.extent(4), (size_t)arg_in.extent(4))),
+        n5(std::min((size_t)arg_out.extent(5), (size_t)arg_in.extent(5))),
+        n6(std::min((size_t)arg_out.extent(6), (size_t)arg_in.extent(6))),
+        n7(std::min((size_t)arg_out.extent(7), (size_t)arg_in.extent(7))) {
+    using Policy = Kokkos::RangePolicy<ExecSpace>;
```

Comment:
Would `MDRangePolicy` work? It requires a `Rank<N>` template argument, where the point of `DynRankView` is to have that vary dynamically.

Also, I just noticed that `n7` isn't used in `operator()`

Reply:
You can explicitly loop on the last two dimensions

Reply:
Why not just get the underlying normal Kokkos::View and then call deep copy/remap on that? I.e. use DownCast. Actually looking at DownCast, we need to fix those functions since they blithely reinterprete_cast the memory of a DynRankView object into a Kokkos::View object which I am not sure is even legal anymore lol. 

But that would solve this issue, then you can just get the appropriate subviews and simply call 
```
Kokkos::deep_copy(exec_space,common_out,common_in);
```